src/Test.sol

```diff
@@ -448,8 +449,471 @@ abstract contract Test is DSTest, Script {
             assertApproxEqRel(a, b, maxPercentDelta);
         }
     }
+
+    /*//////////////////////////////////////////////////////////////
+                              JSON PARSING
+    //////////////////////////////////////////////////////////////*/
+
+   struct RawEIP1559Transaction {
+        string[] arguments;
+        address contractAddress;
+        string contractName;
+        // json value name = function
+        string functionName;
+        bytes32 hash;
+        // json value name = tx
+        RawEIP1559TransactionDetail detail;
+        // json value name = type
+        string contractType;
```

Comment:
Maybe an issue for foundry repo, but I think we should change the JSON names. `function` and `type` are keywords in JS/TS too so having them as field names doesn't seem ideal.

- I like `functionName`
- I'd propose `txDetail` or `txInfo` instead of just `detail`
- `contractType` should be `txType` (e.g. type 0, 1, or 2) IIUC

Reply:
@odyslam saw you marked this resolved—did we change the JSON outputs upstream already or should we create an issue to track?

Reply:
So, we don't **have** to follow the naming convention of foundry. I propose to have the best UX here and then work out to upstream to foundry, but we shouldn't block it until it's upstreamed.

Moreover, a lot of naming changes originate from ethers-rs. At some point I need to change some things there, including the `omit if none` behaviour, which could potentially break the integration here.

Reply:
Ah, yea totally agree we shouldn't block on that—just wanted to make sure we don't lose track of the potential upstream change. I'll create a foundry issue to track and move discussion of potential name changes there, thanks!

Reply:
As I'm creating this, I just noticed it seems the field is called `transaction` in the JSON, not `tx`

Reply:
Also, looks like `function` is the full signature, e.g. `transfer(address,uint256)`, not just the name, so we should rename that to `functionSig` (or `functionSignature`) here

Reply:
Added both of these to the pre-merge tracker comment in https://github.com/foundry-rs/forge-std/pull/138#issuecomment-1229496911, lmk what you think